plugins/destination/postgresql/client/transformer.go

```diff
@@ -80,6 +85,26 @@ func transformArr(arr arrow.Array) []any {
 				Int64: a.Value(i),
 				Valid: a.IsValid(i),
 			}
+		case *array.Uint8:
+			pgArr[i] = pgtype.Int2{
+				Int16: int16(a.Value(i)),
+				Valid: a.IsValid(i),
+			}
+		case *array.Uint16:
+			pgArr[i] = pgtype.Int2{
+				Int16: int16(a.Value(i)),
+				Valid: a.IsValid(i),
+			}
+		case *array.Uint32:
+			pgArr[i] = pgtype.Int4{
+				Int32: int32(a.Value(i)),
+				Valid: a.IsValid(i),
+			}
+		case *array.Uint64:
+			pgArr[i] = pgtype.Int8{
+				Int64: int64(a.Value(i)),
+				Valid: a.IsValid(i),
+			}
```

Comment:
I think here we should try and avoid loss of precision as far as possible: 

 - Uint8 -> cast to int16, store in int2
 - Uint16 -> cast to int32, store in int4
 - Uint32 -> cast to int64, store in int8
 - Uint64 -> store as NUMERIC maybe? 

This way we're doing the best we can within the restrictions imposed by Postgres, and can guarantee no data loss will occur

Reply:
I'm here to remind that we had the same issue in MSSQL migration to Apache Arrow and decided to use upscaling where possible (uin8 -> int16, etc)

Reply:
Yeah, I think we should upscale here, like we did in MSSQL. This way queries will work as expected and there won't be gotchas for users where numbers they expect to always be positive (color values, enums, etc) can be negative if they don't cast it correctly. The cost is using a few extra bytes in the database, but I think 99.9% of users will be fine with that trade-off.

Reply:
I seee we're using `numeric` in other places now, but the cast to `int64` should then be removed here